modules/SofaGeneralObjectInteraction/AttachConstraint.inl

```diff
@@ -268,17 +240,17 @@ void AttachConstraint<DataTypes>::reinit()
     // Check coherency of size between indices vectors 1 and 2
     if(f_indices1.getValue().size() != f_indices2.getValue().size())
     {
-        msg_error() << "Size mismatch between indices1 and indices2 ("
-                    << f_indices1.getValue().size() << " != " << f_indices2.getValue().size() << ").";
+        msg_warning() << "Size mismatch between indices1 and indices2 ("
+                      << f_indices1.getValue().size() << " != " << f_indices2.getValue().size() << ").";
     }
 
     // Set to the correct length if dynamic, else check coherency.
-    if(d_constraintFactor.isSet())
+    if(d_constraintFactor.getValue().size())
     {
-        helper::vector<Real>& constraintFactor = *d_constraintFactor.beginEdit();
+        helper::ReadAccessor<Data<helper::vector<Real>>> constraintFactor = d_constraintFactor;
         if(constraintFactor.size() != f_indices2.getValue().size())
         {
-            msg_error() << "Size of vector constraintFactor, do not fit number of indices attached (" << constraintFactor.size() << " != " << f_indices2.getValue().size() << ").";
+            msg_warning() << "Size of vector constraintFactor, do not fit number of indices attached (" << constraintFactor.size() << " != " << f_indices2.getValue().size() << ").";
```

Comment:
Not having an error can lead to crash if the the number of indices isn't the same